packages/video_player/lib/video_player.dart

```diff
@@ -184,8 +194,15 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   /// is constructed with.
   final DataSourceType dataSourceType;
 
-  static int maxCacheSize = 100 * 1024 * 1024;
-  static int maxCacheFileSize = 10 * 1024 * 1024;
+  /// The maximum cache size to keep on disk in bytes.
+  ///
+  /// eg, 100MB is `100 * 1024 * 1024`
+  final int maxCacheSize;
+
+  /// The maximum size of each individual file in bytes.
+  ///
+  /// eg, 10MB is `10 * 1024 * 1024`
+  final int maxCacheFileSize;
```

Comment:
Why did you change this to be non-static? The backend cache size doesn't change after it has been first created, so I feel like keeping these static creates less problems. Otherwise it might be assumed that the cache sizes can be changed with every newly created `VideoPlayerController`.

Reply:
I wasn't aware that each VideoPlayerController shared the same cache, although it does make sense. Consumers will need to be able to configure this outside of the package source though, so how about we add a comment that states that the cache size is chosen based on the first call? Also, you had it setup so that the only way to turn off the cache was to set cache size to zero, so that's how I decided it was alright to configure on/off with cache size.

Reply:
Adding a comment would also work. I had intended for consumers to set the static variables at some point during app startup, but you're right, that would make it non-intuitive to turn off caching. Maybe it'd be possible to change the cache sizes dynamically though? I'm not sure if the backends support something like that though.
Files and assets aren't cached anyway as the backends only cache if they're passed a http or https URI.

Reply:
Files wouldn't even play on iOS until I make this PR, fwiw. My solution was to pipe non-cached requests to the old video playing mechanism. I make no judgement on if this is the best way forward, though!

Reply:
Honestly I don't need to do anything but set the cache size once and be able to play File/Asset reliably. This PR achieves that for me personally but I agree it should be more clear to consumers what's going on under the hood.

Reply:
Ah okay, then I guess I missed that check on iOS. That part of the PR makes sense and looks good :)
The API for cache sizes doesn't really reflect what's going on though. I still think that the static variables should be used for the sizes. Maybe it'd make sense to just use the `bool isCached` argument?

Reply:
But then you lose the ability to configure the cache size. I think the problem here is iOS (maybe Android?) use singleton pattern for the cache but the package does not. So I think it's okay to just add a comment that the first cache size called is the one that gets used for the entire app.

Reply:
Why do you lose that ability? A consumer can just set these variables once in during their app's startup instead of making sure to always pass non-zero values to `VideoPlayerController`.

Both backends use a singleton cache, maybe it would also be useful to add a new platform method call for this instead of always passing the cache sizes on each `create` call, but that would be another thing.

Reply:
Are you imagining maybe a static method on VideoPlayerController like .setCacheSize? I can see that working.

Reply:
Hm a static method would work as well and would be cleaner to be honest. Static variables were what I was using before this PR, they're not constants. These two variables:
```dart
  static int maxCacheSize = 100 * 1024 * 1024;
  static int maxCacheFileSize = 10 * 1024 * 1024;
```
They could be set by the consumer using `VideoPlayerController.maxCacheSize = 2000 * 1024 * 1024` for example if someone needs a particularly large cache.
What I meant was a new call over the platform channel to the platform-specific implementation, though. That way we also change the communication between Dart and platform code to be more clear. Still, this is something for a future change.

Yeah, that's true. It does all it needs to do. I'm still not sure that passing cache sizes each time to the `VideoPlayerController` constructor is better than the static variables I previously used.

Reply:
Honestly this is a pattern I've never seen before and didn't even know this was possible. It looks like you're correct that it would work as you intend, but it is an informal pattern that may not be expected by Dart consumers.

```
main(){
  print(Foo.number); /// 1
  
  Foo.number = 2;
  
  print(Foo.number); /// 2
  
  final foo = Foo();
  
  print(foo.currentNumber); /// 2
}

class Foo {
  static int number = 1;
  
  int get currentNumber => number;
}
```

Reply:
I mean, setting the static variable doesn't have any side effects in itself. It only changes what's being read later on. I don't know the Dart ecosystem enough to know which patterns are usually used and which aren't though, so if you say that this is an unexpected way to do things, then maybe static variables shouldn't be used. I still would like some way for a consumer to set the cache sizes just once and not pass them to every single constructor call. Maybe with a static method instead to make it at least a bit more clear how it's intended to be used?
Turning off the cache could still be realised using the `enableCache` flag you've added in this PR instead of always giving the option to pass new cache sizes. The current change also has the problem that a consumer could call the `VideoPlayerController` constructor with cache sizes and `enableCache = false` because they don't want to cache this one Uri.

Reply:
My vote would be for a "setCache" static method and back it up internally with static variables and have it throw an error if it's called twice. Thoughts?

Reply:
I just posted one up that I haven't fooled around with yet.

Reply:
That looks pretty good, I like the idea of actually throwing an error there!
I guess now all that remains is to remove the two cache size parameters in the `VideoPlayerController.network` constructor and then we're good.

Reply:
Woops! Nice catch. Fixed that and added some asserts on setCacheSize

Reply:
Okay I think it's good now! Thanks :)

Reply:
Yay!